packages/components/src/List/ListComposition/TextFilter/TextFilter.component.test.js

```diff
@@ -46,64 +46,68 @@ describe('TextFilter', () => {
 			textFilter: '',
 			setTextFilter: jest.fn(),
 		};
+		const event = { target: { value: 'my-filter-value' } };
 
 		// when
 		let wrapper;
 		act(() => {
 			wrapper = mount(
 				<ListContext.Provider value={context}>
-					<TextFilter id="myTextFilter" initialDocked />
+					<TextFilter id="myTextFilter" initialDocked debounceTimeout={0} />
 				</ListContext.Provider>,
 			);
 
-			wrapper
-				.find('FilterBar')
-				.at(0)
-				.prop('onToggle')();
-			wrapper
-				.find('FilterBar')
-				.at(0)
-				.prop('onFilter')({}, 'my-filter-value');
+			wrapper.find('button#myTextFilter').simulate('click');
 		});
 		wrapper.update();
+		wrapper.find('input').at(0).simulate('change', event);
```

Comment:
Refacto the test to make it more futureproof. We don't select the components but the dom element instead, and simulate events instead of calling the function from props